tests/cases/fourslash/autoImportNodeNextJSRequire.ts

```diff
@@ -0,0 +1,25 @@
+/// <reference path='fourslash.ts' />
+
+// @module: nodenext
+// @allowJs: true
+// @checkJs: true
+// @noEmit: true
+
+// @Filename: /matrix.js
+//// exports.variants = [];
+
+// @Filename: /main.js
+//// exports.dedupeLines = data => {
+////   variants/**/
+//// }
+
+// @Filename: /totally-irrelevant-no-way-this-changes-things-right.js
+//// export default 0;
```

Comment:
We were using the presence of ES module syntax in other files as an indicator of what to use, but `--module nodenext` is a stronger signal—we should default to syntax that won’t crash in Node.